### PR TITLE
Update Repository Readme

### LINKED ISSUE #3177: Make translation instructions easier to find
Currently, the only locations where a link to instructions about how to help translate are the "Help Translate" button in the Scratch Addons settings and within the docs themselves. This makes it relatively hard to find and even harder to stumble across.

A couple of potential places to put it:
- Under the language dropdown on scratchaddons.com
- In the description for "International" (under the "Features" header) on scratchaddons.com
- In the footer on scratchaddons.com
- In the "Contribute" section of the README in the ScratchAddons repo
- In the "Contributing guidelines" (`CONTRIBUTING.md`) in the ScratchAddons repo
- Some/all of the above

### LINKED ISSUE #5411: Remove mention of persistent scripts from README
I know that Manifest V3 is going to make persistent scripts harder to manage, and that you removed the docs page on creating them earlier this year, but they're still mentioned in this repository's `README.md`. Should we remove the mention of them there too?

### PR DESCRIPTION
**Pull request by Samq64**
Resolves #3177
Resolves #5411
Closes #5603

Previous PR: #6208

### Changes

Makes some significant changes to the repo README.

### Reason for changes

The README needs an update.

### Tests

https://github.com/Samq64/ScratchAddons/tree/update-readme2#readme

### CODE REVIEW

**Comment by apple502j** [comment]
Images should be placed inside `.github`.

**Comment by Samq64** [comment]
I think the content looks really good now. I just need to find a way to make the top look a little better. The problem with moving the installation to the top is I want to keep the entire about section together and I think it's more important than the installation instructions, but I also want to keep the badges on top for quick access for users who just want to install the store versions. Maybe the second row of badges could be removed though.

**Comment by Samq64** [comment]
I added the pending label because the installation instructions mention building the settings page now.

File: README.md [hunk 1]
```diff
@@ -13,19 +13,19 @@
 [![](https://img.shields.io/badge/chat-on_discord-7289da.svg?style=flat-square)](https://discord.gg/R5NBqwMjNc)
 [![](https://img.shields.io/badge/website-scratchaddons.com-ff7b26.svg?style=flat-square)](https://scratchaddons.com)
 
-<img src="images/screenshots/demo.png" alt="Scratch Addons demo" height=500px></img>
+<img src=".github/readme-images/demo.png" alt="Scratch Addons demo" height=500px></img>
 
 ## What is Scratch Addons?
 
-[Scratch Addons](https://github.com/ScratchAddons) combines new and existing features and themes for the Scratch website and project editor into one easy-to-access and configurable browser extension. The mission is to provide a centralized, up-to-date platform for community development of new features and themes for Scratch.
+Scratch Addons combines new and existing features and themes for the [Scratch](https://github.com/scratchfoundation) website and project editor into one easy-to-access and configurable browser extension. The mission is to provide a centralized, up-to-date platform for community development of new features and themes for Scratch.
```
Comment (CST1229) [inline README.md:18]:
```suggestion
Scratch Addons combines new and existing features and themes for the [Scratch](https://scratch.mit.edu) website and project editor into one easy-to-access and configurable browser extension. The mission is to provide a centralized, up-to-date platform for community development of new features and themes for Scratch.
```

Comment (WorldLanguages) [reply README.md:18]: Maybe https://scratch.mit.edu/about if you prefer.

File: README.md [hunk 2]
```diff
@@ -1,37 +1,39 @@
...
+### What is an addon?
 
-Userscripts and persistent scripts can use the `addon.*` JavaScript APIs, which allow them to obtain Scratch-related information (for example, get the current logged in user) and also use extension APIs (like sending notifications).
+An addon is one or more [userscripts](https://scratchaddons.com/docs/develop/userscripts/) or [userstyles](https://scratchaddons.com/docs/develop/userstyles/) that run on the Scratch website or project editor.
 
-Converting an already existing extension/userscript into an addon, or writing your own, is very easy. [Check out the guide](https://scratchaddons.com/docs/develop/getting-started/creating-an-addon/).
+Userscripts can use the `addon.*` JavaScript APIs provided by Scratch Addons, which allows them to obtain Scratch-related information like the currently logged in user and use extension APIs to do things such as send notifications.
```
Comment (WorldLanguages) [inline README.md:30]: I must admit the `addon.* APIs` description on the docs is quite old. Only one addon sends notifications and not through an addon.* API.

```suggestion
Userscripts can also use the `addon.*` JavaScript APIs provided by Scratch Addons, which make it easy to obtain Scratch-related information like the currently logged in user, wait until a certain element exists on the page, or get a reference to the Scratch VM object.
```

File: README.md [hunk 3]
```diff
@@ -1,37 +1,39 @@
...
 
-### If everything is an addon, then what does Scratch Addons do?
+Read the [documentation](https://scratchaddons.com/docs/develop/getting-started/addon-basics/) for more information about addons.
 
-Scratch Addons by itself is just an addon loader. Its main tasks are:
+### If everything is an addon, what does Scratch Addons do?
+
+Scratch Addons by itself is just an addon loader. Its main tasks are to:
```
Comment (WorldLanguages) [inline README.md:36]: I would remove this whole section until we have better docs on non-addon code, in that case we'll just link to them.

File: README.md [hunk 4]
```diff
@@ -1,37 +1,39 @@
...
+
+<div align="center">
+
+![Scratch Addons screenshot](.github/readme-images/screenshot.png)
 
-### What's actually an "addon"?
+</div>
 
-An addon is similar to an extension or a userscript, but they use special APIs provided by the Scratch Addons extension. These APIs allow addons to run scripts on a Scratch page (userscripts), run scripts on the background (persistent scripts), or apply styles to the Scratch website (userstyles).
+### What is an addon?
```
Comment (WorldLanguages) [inline README.md:26]:
```suggestion
### About addons
```

File: README.md (hunk 3 above)
Comment (WorldLanguages) [reply README.md:36]: Maybe this section could be renamed to `About the browser extension` and it can be made shorter.

File: README.md [hunk 5]
```diff
@@ -43,41 +45,73 @@ Scratch Addons by itself is just an addon loader. Its main tasks are:
 
 ### Addons outside of Scratch Addons itself
```
Comment (WorldLanguages) [inline README.md:46]: Maybe this could be rephrased? Something like:

```suggestion
### Addons beyond the Scratch Addons browser extension
```

File: README.md (hunk 2 above)
Comment (WorldLanguages) [reply README.md:30]: Even better if you modify my suggestion to imply the addon.* API can do many more things than the three ones I mentioned.

File: README.md (hunk 3 above)
Comment (WorldLanguages) [reply README.md:36]: Maybe we could go straight to the point: the browser extension only runs addons when necessary (if the URL matches), it provides addon.* APIs and provides a settings page to the user.

**Comment by WorldLanguages** [comment]
In the `What is an addon?` section (or `About addons`), we could mention some basic values that we take for granted: addons should be compatible with each other, be optimized to avoid hurting performance (specially in the editor), be localizable, accessible (which includes color contrast and RTL support), have good browser support, preserve user privacy, etc.
We should also mention addon manifests, not just the `addon.* APIs`.

File: README.md [hunk 6]
```diff
@@ -1,37 +1,39 @@
...
+![Scratch Addons screenshot](.github/readme-images/screenshot.png)
 
-### What's actually an "addon"?
+</div>
 
-An addon is similar to an extension or a userscript, but they use special APIs provided by the Scratch Addons extension. These APIs allow addons to run scripts on a Scratch page (userscripts), run scripts on the background (persistent scripts), or apply styles to the Scratch website (userstyles).
+### What is an addon?
 
-Userscripts and persistent scripts can use the `addon.*` JavaScript APIs, which allow them to obtain Scratch-related information (for example, get the current logged in user) and also use extension APIs (like sending notifications).
+An addon is one or more [userscripts](https://scratchaddons.com/docs/develop/userscripts/) or [userstyles](https://scratchaddons.com/docs/develop/userstyles/) that run on the Scratch website or project editor.
```
Comment (WorldLanguages) [inline README.md:28]:
```suggestion
An addon is one or more [userscripts](https://scratchaddons.com/docs/develop/userscripts/) (written in JavaScript) or [userstyles](https://scratchaddons.com/docs/develop/userstyles/) (written in CSS) that run on the Scratch website or project editor.
```

File: README.md [hunk 7]
```diff
@@ -41,43 +43,75 @@ Scratch Addons by itself is just an addon loader. Its main tasks are:
 - Avoid addons from interfering with each other.
 - Avoid duplicate work from different addons.
 
-### Addons outside of Scratch Addons itself
+### Addons beyond the Scratch Addons browser extension
+
+Other extensions and even forks of Scratch can also provide their users with most addons from Scratch Addons, as long as a compatibility layer for `addon.*` and other parts of the addon loader are present. A notable example is the [TurboWarp](https://github.com/TurboWarp/scratch-gui/tree/develop/src/addons) editor. These also inherit translations from the Scratch Addons project.
```
Comment (WorldLanguages) [inline README.md:48]: We could mention adacraft (TW fork)

Comment (Samq64) [reply README.md:48]: We link to the source for TurboWarp. Should we just to the site or editor for both?

File: README.md [hunk 8]
```diff
@@ -1,83 +1,120 @@
...
-[![](https://img.shields.io/amo/users/scratch-messaging-extension?style=flat-square&logo=firefox-browser&logoColor=white&label=users&color=FF7139)](https://addons.mozilla.org/firefox/addon/scratch-messaging-extension/)
-[![](https://img.shields.io/github/v/release/ScratchAddons/ScratchAddons?style=flat-square&logo=github&logoColor=white&label=version&color=181717)](https://github.com/ScratchAddons/ScratchAddons/releases)
-[![](https://img.shields.io/github/downloads/ScratchAddons/ScratchAddons/total?style=flat-square&logo=github&logoColor=white&label=downloads&color=181717)](https://github.com/ScratchAddons/ScratchAddons/releases)
-[![](https://img.shields.io/badge/discuss-on_github-181717.svg?style=flat-square)](https://github.com/ScratchAddons/ScratchAddons/discussions)
+<img src="images/icon.svg" alt="Scratch Addons logo" align="right" width="128px"></img>
+# Scratch Addons Browser Extension
+
+[![Chrome Web Store](.github/readme-images/cws-badge.png)](https://chrome.google.com/webstore/detail/fbeffbjdlemaoicjdapfpikkikjoneco)
+[![Firefox Add-ons](.github/readme-images/ff-addon-badge.png)](https://chrome.google.com/webstore/detail/fbeffbjdlemaoicjdapfpikkikjoneco)
```
Comment (DNin01) [inline README.md:5]: Looks like both of these links go to the Chrome Web Store.
```suggestion
[![Firefox Add-ons](.github/readme-images/ff-addon-badge.png)](https://addons.mozilla.org/firefox/addon/scratch-messaging-extension/)
```

**Review by DNin01 — DISMISSED** [review]

**Review by DNin01 — COMMENTED** [review]
Seems okay to me. I'll review more thoroughly before I approve.

File: README.md [hunk 9]
```diff
@@ -1,83 +1,120 @@
...
-[![](https://img.shields.io/amo/v/scratch-messaging-extension?style=flat-square&logo=firefox-browser&logoColor=white&label=version&color=FF7139)](https://addons.mozilla.org/firefox/addon/scratch-messaging-extension/)
-[![](https://img.shields.io/amo/users/scratch-messaging-extension?style=flat-square&logo=firefox-browser&logoColor=white&label=users&color=FF7139)](https://addons.mozilla.org/firefox/addon/scratch-messaging-extension/)
-[![](https://img.shields.io/github/v/release/ScratchAddons/ScratchAddons?style=flat-square&logo=github&logoColor=white&label=version&color=181717)](https://github.com/ScratchAddons/ScratchAddons/releases)
-[![](https://img.shields.io/github/downloads/ScratchAddons/ScratchAddons/total?style=flat-square&logo=github&logoColor=white&label=downloads&color=181717)](https://github.com/ScratchAddons/ScratchAddons/releases)
-[![](https://img.shields.io/badge/discuss-on_github-181717.svg?style=flat-square)](https://github.com/ScratchAddons/ScratchAddons/discussions)
+<img src="images/icon.svg" alt="Scratch Addons logo" align="right" width="128px"></img>
+# Scratch Addons Browser Extension
```
Comment (Hans5958) [inline README.md:2]: For the not-sure-how-many times here, I would still suggest this.

```suggestion
# Scratch Addons
```

**Review by Hans5958 — APPROVED** [review]

File: README.md (hunk 9 above)
Comment (Hans5958) [reply README.md]: Just want to tell that I don't intend malice or being forceful here. If you think that's the case, I don't mean to. Sorry.

Comment (Samq64) [reply README.md]: I didn't mean to sound rude.

Comment (WorldLanguages) [reply README.md]: It's okay, if someone cares enough about changing the title they can make a dedicated PR for that after we merge this one

File: README.md [hunk 10]
```diff
@@ -1,83 +1,120 @@
...
+
+Userscripts work similarly to [extension content scripts](https://developer.chrome.com/docs/extensions/content_scripts/), but they also have access to `addon.*` APIs. Userscripts can use these built-in utilities for various purposes: waiting until a certain element exists on the page, listening to URL change events, getting a reference to the Scratch VM object, etc.
+
+Addons are designed to be compatible with each other. They are also developed with performance, internationalization, accessibility, and privacy in mind.
+
+Read the [documentation](https://scratchaddons.com/docs/develop/getting-started/addon-basics/) for more information about addons.
+
+### About the extension
+
+The Scratch Addons browser extension provides a settings page where users can enable, disable and configure addon settings. The extension interprets addon manifests, stores the user's settings, and provides APIs to userscripts. A new version of the extension is sent to stores at least once a month.
```
Comment (DNin01) [inline README.md:40]: Might I suggest rewording the part where a new version is released at least once a month? I would also use the word "release" and mention how each major update releases with new features.

Comment (WorldLanguages) [reply README.md:40]: Go ahead

**Review by DNin01 — APPROVED** [review]
All well written. Good job!

**Comment by WorldLanguages** [comment]
This should be pretty good to go if we remove the parts that mention building the `webpages` folder, as most likely we'll merge this first.

**Review by WorldLanguages — APPROVED** [review]
🎉

File: README.md [hunk 11]
```diff
@@ -1,83 +1,109 @@
...
+- `addon-api`: Implementation of the `addon.*` JavaScript APIs.
+- `addons`: Each addon has its own directory, which must include an addon manifest file named `addon.json`.
+- `libraries`: Third-party libraries and other utilities, some of which are used by addons.
+
+#### Others
+- `.github`: GitHub templates, workflows and contributing files.
+- `_locales`: Translation strings for the browser extension (excludes addons).
+- `background`: Background scripts for the extension.
+- `content-scripts`: Content scripts, which among other things, execute userscripts and inject userstyles to the page.
+- `images`: Logos, screenshots and icons. Addon-specific images belong in the addon's directory instead.
```
Comment (apple502j) [inline README.md:59]: Here as well

File: README.md [hunk 12]
```diff
@@ -1,83 +1,109 @@
...
+- `popups`: Addon pages that are only accessible through the extension popup (for example, Scratch Messaging).
+- `webpages`: The settings page, extension popup, and other pages.
 
-### What's actually an "addon"?
+## Installation
 
-An addon is similar to an extension or a userscript, but they use special APIs provided by the Scratch Addons extension. These APIs allow addons to run scripts on a Scratch page (userscripts), run scripts on the background (persistent scripts), or apply styles to the Scratch website (userstyles).
+Building is not required unless working with the `webpages` directory. The best way to download the source is with Git:
```
Comment (apple502j) [inline README.md:65]: We do not build webpages for now.

File: README.md [hunk 13]
```diff
@@ -1,83 +1,109 @@
...
-- Run addons and provide APIs to them.
-- Provide global state to addons (for example, the `addon.auth` API).
-- Pollute prototypes for use by addon userscripts.
-- Provide ways to access and modify Redux state.
-- Avoid addons from interfering with each other.
-- Avoid duplicate work from different addons.
+### Loading the extension (Firefox)
 
-### Addons outside of Scratch Addons itself
+Go to `about:debugging`, select "This Firefox", click "Load Temporary Add-on..." and select the `manifest.json` file in the `ScratchAddons` folder.
```
Comment (apple502j) [inline README.md:79]: oxford comma

**Review by apple502j — CHANGES_REQUESTED** [review]

File: README.md [hunk 14]
```diff
@@ -1,83 +1,109 @@
...
+
+</div>
+
+### About addons
+
+An addon mainly consists of one or more [userscripts](https://scratchaddons.com/docs/develop/userscripts/) (written in JavaScript) or [userstyles](https://scratchaddons.com/docs/develop/userstyles/) (written in CSS) that run on the Scratch website or project editor.
+
+Each addon declares its own [addon manifest](https://scratchaddons.com/docs/reference/addon-manifest/) (`addon.json` file). This file specifies under which circumstances each one of its userscripts and userstyles should be injected into the page. It also contains user-facing information, such as the description of the feature, and more information about the addon's settings.
+
+Userscripts work similarly to [extension content scripts](https://developer.chrome.com/docs/extensions/content_scripts/), but they also have access to `addon.*` APIs. Userscripts can use these built-in utilities for various purposes: waiting until a certain element exists on the page, listening to URL change events, getting a reference to the Scratch VM object, etc.
```
[the remark this answers is not in the document]
Comment (WorldLanguages) [reply README.md]: I may try to rewrite this but keeping the content scripts parallelism.

**Review by apple502j — APPROVED** [review]
